src/types/uint-16.ts

```diff
@@ -1,12 +1,39 @@
-import { makeClass } from "../utils/make-class";
 import { UInt } from "./uint";
 
-const UInt16 = makeClass(
-  {
-    inherits: UInt,
-    statics: { width: 2 },
-  },
-  undefined
-);
+/**
+ * Derived UInt class for serializing/deserializing 16 bit UInt
+ */
+class UInt16 extends UInt {
+  static readonly width: number = 2
```

Comment:
optional: I think this is pretty obvious but *maybe* a bit more obvious to write something like:
```
static readonly width: number = 16 / 8 // 2
```